buildbot/check.py

```diff
@@ -1,24 +1,41 @@
 import argparse
 import os
+import multiprocessing
 import subprocess
 import sys
 
 DEFAULT_CPU_COUNT = 4
 
 def do_check(args):
-    ret = False
-
-    cpu_count = os.cpu_count()
-    if cpu_count is None:
+    try:
+        cpu_count = multiprocessing.cpu_count()
+    except NotImplementedError:
         cpu_count = DEFAULT_CPU_COUNT
 
+    # Get absolute path to source directory
+    if args.src_dir:
+      abs_src_dir = os.path.abspath(args.src_dir)
+    else:
+      abs_src_dir = os.path.abspath(os.path.join(__file__, "../.."))
+    # Get absolute path to build directory
+    if args.obj_dir:
+      abs_obj_dir = os.path.abspath(args.obj_dir)
+    else:
+      abs_obj_dir = os.path.join(abs_src_dir, "build")
+
+    cmake_cmd = [
+        "cmake",
+        "--build", "{}".format(abs_obj_dir),
```

Comment:
Could you elaborate on this suggestion, please?

Reply:
Sure, in short, ```"{}".format(abs_obj_dir)``` do the following: constructs a new string and initializes it with placeholder ```{}```. This placeholder initializes with the value of argument of ```format()``` method, which is ```abs_obj_dir```. As list ```cmake_cmd``` which passes to ```check_call``` function is the list of strings, we can add ```abs_obj_dir``` directly to this list, because it is regular string, w/o constructing a new string and copying one string to this new string

Reply:
Okay. Let's try your suggestions.